src/legacy/ui/public/filter_bar/query_filter.js

```diff
@@ -370,11 +380,9 @@ export function FilterBarQueryFilterProvider(Private, Promise, indexPatterns, $r
 
         // save states and emit the required events
         saveState();
-        queryFilter.emit('update')
-          .then(function () {
-            if (!doFetch) return;
-            queryFilter.emit('fetch');
-          });
+        update$.next();
+        if (!doFetch) return;
+        fetch$.next();
```

Comment:
The early return is fine in the old code because it ends an inner function.

```js
function() {
  if (!doFetch) return;
  // ...
}
```

But in new code the early return would end the outer function (which is fine as there is nothing else in this function, but is a bit confusing), maybe write it like this:

```js
if (doFetch) fetch$.next();
```

Also, I'm not sure why `.emit('fetch')` is inside `.then()` in the old code. Any ideas?

Reply:
Agreed.

Not sure as well about this implementation detail ATM, but wanted to preserve behavior.

Reply:
Fixed the return issue